regolith/builders/beamplanbuilder.py

```diff
@@ -0,0 +1,98 @@
+"""Builder for Current and Pending Reports."""
+from regolith.builders.basebuilder import LatexBuilderBase
+from regolith.tools import all_docs_from_collection
+from regolith.tools import group
+import pandas as pd
+
+
+class BeamPlanBuilder(LatexBuilderBase):
+    """
+    Build a file of experiment plans for the beamtime from database entries. The report is in .tex file. The template
+    of the file is in the 'templates/beamplan.txt'. The data will be grouped according to beamtime. Each beamtime
+    will generate a file of the plans. If 'beamtime' in 'rc' are not None, only plans for those beamtime will be
+    generated.
+    """
+    btype = "beamplan"
+    needed_dbs = ['beamplan', "people"]
+
+    def construct_global_ctx(self):
+        """Constructs the global context"""
+        super().construct_global_ctx()
+        gtx = self.gtx
+        rc = self.rc
+        gtx["beamplan"] = all_docs_from_collection(rc.client, "beamplan")
+        gtx["all_docs_from_collection"] = all_docs_from_collection
+        gtx["float"] = float
+        gtx["str"] = str
+
+    @staticmethod
```

Comment:
why did you choose to make it a static method?

Reply:
Because it doesn't use any attribute or method in the class but I think this method is still strictly associated with the class.

Reply:
Also, I decide to make it a private method. I don't think people will use it and I don't want others to touch it since the amendment of it will break the building.